### ISSUE TITLE
user or password

### ISSUE THREAD

**Issue by ghost** [dguo/pawblock#8]
you can add the option to create a user or password to enter your add-on and that it is not deactivated, or that for the add-on to be disabled ask for a confirmation password

**Comment by dguo** [comment]
Hi, thanks for the suggestion.

Is the motivation to stop someone else from turning off the add-on? One issue is that the user can always just remove the add-on through the browser's settings. The add-on can't do anything about that.

**Comment by ghost** [comment]
yes, it is so that someone else or myself can not disable the add-on unless
that write the password

**Comment by dguo** [comment]
It would be possible to add a password for turning the functionality off from within the extension, but the user could always disable or uninstall the extension from the browser's settings or by right clicking the extension.

Browsers [don't allow](https://stackoverflow.com/a/29825932/1481479) extensions to stop themselves from being uninstalled.

**Comment by Tamono** [comment]
I have seen a blocking extension that opened a password prompt when trying to enter about:addons.
The only way around it was opening firefox in safe mode without addons, I think it was before quantum, though. 

One could argue, however, that users should show at least some discipline themselves because otherwise you will always find a way to fool yourself. 😄
